packages/passport-auth/src/index.ts

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+import * as passport from 'passport';
+import {Strategy} from 'passport-local';
+import * as express from 'express';
+
+interface PassportAuth<T> {
+  init(app: express.Express): void
+  setStrategy(strategy: Function): void
+  setSerializeFunction(serialize: Function): void
+  setDeserializeFunction(deserialize: Function): void
+  authenticate(strategy: string, opts: Object): void
+}
+
+class PassportAuthImpl<T> implements PassportAuth<T> {
+  public init(app: express.Express) {
+    console.log('>>>>>>>', 'initializing passport auth');
+    app.use(passport.initialize());
+    app.use(passport.session());
+  }
+
+  public setStrategy(strategy: Function) {
+    console.log('>>>>>>>', 'setting strategy');
+    passport.use(new Strategy((username: string, password: string, done: Function) => {
+      strategy(username, password, done);
+    }));
+  }
+
+  public setSerializeFunction(serialize: Function) {
+    //determines which data of the user object should be stored in the session.
+    //will be attached to req.session.passport.user
+    //Will use this info to deserialize the whole user object later.
+    console.log('>>>>>>>', 'setting serialize function');
+    passport.serializeUser((user: T, done: Function) => {
+      serialize(user, done);
+    });
+  }
+
+  public setDeserializeFunction(deserialize: Function) {
+    //Retrieves user object from the source using the id as an identifier.
+    console.log('>>>>>>>', 'setting deserialize function');
+    passport.deserializeUser((id: string, done: Function) => {
+      deserialize(id, done);
+    });
+  };
+
+  public authenticate(strategy: string, opts: Object) {
```

Comment:
Maybe this could be called `login` instead of `authenticate` to stay inline with keycloak naming scheme. 

We could have just a `login()` action which **just** performs authentication.

Then we can also have the `protect()` action (logic added down the line) which will do a few things:
1. Check if User is Authenticated, if not call `login()`
2. If user is authenticated, call `hasResourceRole(roleOnTheRoute)`
3. If user hadn't the role deny access, if they have the role, continue.